scripts/upgrade/TimeConstraints.sol

```diff
@@ -30,39 +36,69 @@ contract TimeConstraints {
-            revert InvalidDayTimeRange(startDayTime, endDayTime);
+            if (currentDayTime < startDayTime && currentDayTime > endDayTime) {
+                revert DayTimeOutOfRange(currentDayTime, startDayTime, endDayTime);
+            }
+        } else {
+            if (currentDayTime < startDayTime || currentDayTime > endDayTime) {
+                revert DayTimeOutOfRange(currentDayTime, startDayTime, endDayTime);
+            }
         }
+    }
 
-        Duration currentDayTime = getCurrentDayTime();
-        if (currentDayTime < startDayTime || currentDayTime > endDayTime) {
-            revert DayTimeOutOfRange(currentDayTime, startDayTime, endDayTime);
-        }
+    /// @notice Checks that the current day time satisfies specific time range during the day and emits an event.
+    /// @dev Supports two types of time ranges:
+    ///      1. Regular range: startDayTime <= endDayTime (e.g. [12:00, 18:00])
+    ///      2. Overnight range: startDayTime > endDayTime (e.g. [18:00, 12:00], where the end time is on the next day)
+    /// @param startDayTime The start time of the allowed range in seconds since midnight (UTC).
+    /// @param endDayTime The end time of the allowed range in seconds since midnight (UTC).
+    function checkTimeWithinDayTimeAndEmit(Duration startDayTime, Duration endDayTime) external {
+        checkTimeWithinDayTime(startDayTime, endDayTime);
+        emit TimeWithinDayTimeChecked(startDayTime, endDayTime);
     }
 
-    /// @notice Checks that the transaction can only be executed after a specific timestamp.
+    /// @notice Checks that the current timestamp is after the given specific timestamp.
     /// @param timestamp The Unix timestamp after which the function can be executed.
-    function checkExecuteAfterTimestamp(Timestamp timestamp) external view {
-        if (Timestamps.now() < timestamp) {
+    function checkTimeAfterTimestamp(Timestamp timestamp) public view {
+        if (Timestamps.now() <= timestamp) {
             revert TimestampNotReached(timestamp);
         }
     }
 
-    /// @notice Checks that the transaction can only be executed before a specific timestamp.
+    /// @notice Checks that the current timestamp is after the given specific timestamp and emits an event.
+    /// @param timestamp The Unix timestamp after which the function can be executed.
+    function checkTimeAfterTimestampAndEmit(Timestamp timestamp) external {
+        checkTimeAfterTimestamp(timestamp);
+        emit TimeAfterTimestampChecked(timestamp);
+    }
+
+    /// @notice Checks that the current timestamp is before the given specific timestamp.
     /// @param timestamp The Unix timestamp before which the function can be executed.
-    function checkExecuteBeforeTimestamp(Timestamp timestamp) external view {
-        if (Timestamps.now() > timestamp) {
+    function checkTimeBeforeTimestamp(Timestamp timestamp) public view {
+        if (Timestamps.now() >= timestamp) {
```

Comment:
`TimestampPassed(timestamp)` for error name, to match the `checkAfterTimestamp()` semantic